linux-kernel-vs-memory-fragmentation-2.md

```diff
@@ -0,0 +1,144 @@
+---
+title: Linux Kernel vs. Memory Fragmentation (Part II)
+author: ['Wenbo Zhang']
+date: 2021-05-03
+summary: This post introduces the principle of memory compaction, how to view the fragmentation index, and how to quantify the latency overheads caused by memory compaction.
+tags: ['Linux']
+categories: ['Engineering']
+image: /images/blog/linux-memory-fragmentation-and-defragmentation.png
+---
+
+**Author:** Wenbo Zhang (Linux Kernel Engineer of the EE team at PingCAP)
+
+**Transcreator:** [Charlotte Liu](https://github.com/CharLotteiu); **Editor:** Tom Dewan
+
+![Linux kernel memory fragmentation and defragmentation](media/linux-memory-fragmentation-and-defragmentation.png)
+
+In [Linux Kernel vs. Memory Fragmentation (Part I)](https://en.pingcap.com/blog/linux-kernel-vs-memory-fragmentation-1), I concluded that grouping by migration types only delays memory fragmentation, but does not fundamentally solve it. As the memory fragmentation increases and it does not have enough contiguous physical memory, performance degrades.
+
+Therefore, to mitigate the performance degradation, the Linux kernel community introduced **memory compaction** to the kernel.
+
+In this post, I'll explain the principle of memory compaction, how to view the fragmentation index, and how to quantify the latency overheads caused by memory compaction.
+
+## Memory compaction
+
+Before memory compaction, the kernel used lumpy reclaim for defragmentation. However, this feature was removed from v3.10 (currently the most widely used kernel version). If you’d like to learn more, you can read about lumpy reclaim in the articles I listed in [A brief history of defragmentation](https://en.pingcap.com/blog/linux-kernel-vs-memory-fragmentation-1#a-brief-history-of-defragmentation). For now, let me bring your mind to memory compaction.
```

Comment:
```suggestion
Before memory compaction, the kernel used lumpy reclaim for defragmentation. However, this feature was removed from v3.10 (currently the most widely used kernel version). If you'd like to learn more, you can read about lumpy reclaim in the articles I listed in [A brief history of defragmentation](https://pingcap.com/blog/linux-kernel-vs-memory-fragmentation-1#a-brief-history-of-defragmentation). For now, let me bring your mind to memory compaction.
```

Reply:
Please update all quotation marks to straight and remove "en." from all URLs about the PingCAP website.